sw/snRuntime/src/start.c

```diff
@@ -99,8 +99,12 @@ static inline void snrt_init_libs() { snrt_alloc_init(); }
 #ifdef SNRT_CRT0_EXIT
 static inline void snrt_exit_default(int exit_code) {
     exit_code = snrt_global_all_to_all_reduction(exit_code);
+#ifdef __GVSOC__
+    if (snrt_global_core_idx() == 0) __semihost(0x18, (long)exit_code);
+#else
     if (snrt_global_core_idx() == 0)
         *(snrt_exit_code_destination()) = (exit_code << 1) | 1;
+#endif
```

Comment:
The files in the `sw/snRuntime` folder are meant to be base sources for multiple different Snitch runtime implementations. Different implementations already exist for [RTL](https://github.com/pulp-platform/snitch_cluster/tree/main/target/snitch_cluster/sw/runtime/rtl) and for [Banshee](https://github.com/pulp-platform/snitch_cluster/tree/main/target/snitch_cluster/sw/runtime/banshee).

I wouldn't implement this change here, but rather create a custom runtime for GVSoC, which defines its own exit function.